pydantic/_internal/_known_annotated_metadata.py

```diff
@@ -102,6 +102,22 @@
     CONSTRAINTS_TO_ALLOWED_SCHEMAS[constraint].update(('enum',))
 
 
+def _is_sequence_metadata(schema: CoreSchema) -> bool:
+    """Checks if the schema is a sequence schema.
+
+    Args:
+        schema: The core schema.
+
+    Returns:
+        True if the schema is a sequence schema, False otherwise.
+    """
+    python_schema = schema.get('python_schema', {})
+    steps = python_schema.get('steps', [{}])
+    first_step = steps[0]
+    cls_type = first_step.get('cls')
+    return cls_type == Sequence
```

Comment:
I feel like this is very coupled to the exact form of the `CoreSchema` we use for sequence validation, which may change.

It might be more robust to put some `metadata` in the sequence schema (in `_generate_schema.py`) and check for that here.

Reply:
Yeah, makes sense. So, I've put some extra metadata during the sequence-schema-generation process, and then utilized it, as described, to check if it's of a Sequence type we are looking for. Is there anything more to it?